From 22f08dcac717d53b655b2e766db8e3e0d934b0b2 Mon Sep 17 00:00:00 2001
From: Ishaaq Ollite <admin@eshark.tk>
Date: Sat, 30 Dec 2023 22:04:26 +0000
Subject: [PATCH] Update Makefile.dtbinst

---
 scripts/Makefile.dtbinst | 17 ++++++++++++-----
 1 file changed, 12 insertions(+), 5 deletions(-)

diff --git a/scripts/Makefile.dtbinst b/scripts/Makefile.dtbinst
index 1e48a9d61ee7..49ee30b1d513 100644
--- a/scripts/Makefile.dtbinst
+++ b/scripts/Makefile.dtbinst
@@ -1,4 +1,3 @@
-# SPDX-License-Identifier: GPL-2.0
 # ==========================================================================
 # Installing dtb files
 #
@@ -6,6 +5,8 @@
 # INSTALL_DTBS_PATH directory or the default location:
 #
 #   $INSTALL_PATH/dtbs/$KERNELRELEASE
+#
+# Traverse through subdirectories listed in $(dts-dirs).
 # ==========================================================================
 
 src := $(obj)
@@ -13,21 +14,27 @@ src := $(obj)
 PHONY := __dtbs_install
 __dtbs_install:
 
-export dtbinst_root ?= $(obj)
+export dtbinst-root ?= $(obj)
 
 include include/config/auto.conf
 include scripts/Kbuild.include
 include $(src)/Makefile
 
-dtbinst-files	:= $(sort $(dtb-y) $(if $(CONFIG_OF_ALL_DTBS), $(dtb-)))
+PHONY += __dtbs_install_prep
+__dtbs_install_prep:
+ifeq ("$(dtbinst-root)", "$(obj)")
+	$(Q)mkdir -p $(INSTALL_DTBS_PATH)
+endif
+
+dtbinst-files	:= $(dtb-y)
 dtboinst-files	:= $(dtbo-y)
-dtbinst-dirs	:= $(subdir-y) $(subdir-m)
+dtbinst-dirs	:= $(dts-dirs)
 
 # Helper targets for Installing DTBs into the boot directory
 quiet_cmd_dtb_install =	INSTALL $<
       cmd_dtb_install =	mkdir -p $(2); cp $< $(2)
 
-install-dir = $(patsubst $(dtbinst_root)%,$(INSTALL_DTBS_PATH)%,$(obj))
+install-dir = $(patsubst $(dtbinst-root)%,$(INSTALL_DTBS_PATH)%,$(obj))
 
 $(dtbinst-files) $(dtboinst-files) $(dtbinst-dirs): | __dtbs_install_prep
 
